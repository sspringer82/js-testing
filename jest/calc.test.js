const Calculator = require('./calc');

describe('Calculator', () => {
  let calc;
  beforeAll(() => {});
  beforeEach(() => {
    // Arrange
    calc = new Calculator();
  });
  afterEach(() => {});
  afterAll(() => {});

  describe('add', () => {
    it('should add 1 and 1 and return 2', () => {
      // tripple A
      // Arrange
      // Act
      const result = calc.add(1, 1);
      // Assert
      expect(result).toBe(2);
    });

    it('should match snapshot', () => {
      expect(calc.render(4)).toMatchSnapshot();
    });
  });
});
