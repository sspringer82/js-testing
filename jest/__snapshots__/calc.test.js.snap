// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator add should match snapshot 1`] = `"<div>4</div>"`;
